### DISCUSSION TITLE
Downloading and creating .fit files, no upload to Garmin

### DISCUSSION THREAD

**Discussion by mkblm** [philosowaffle/peloton-to-garmin#419] [Q&A] [answered]
Hello- I downloaded the latest stable package, setup configuration.local.json, and ran PelotonToGarminConsole.exe. Peloton workouts were downloaded as .fit files, but nothing uploaded to Garmin. The command window is just sitting at "Sleeping for 86400 seconds..." End of the log file is below. Any thoughts, am I missing something in configuration? I only see two exe files (the other createdump.exe, so doesn't seem to be a separate one for uploading to Garmin. 

I did try downloading the source code and compiling/executing some individual projects in the solution from there, but ran into some issues doing that so just trying the simplest way. Thanks in advance for any help that someone can offer!

...
2023-01-19 19:27:09.261 -05:00 [INF] [@Format] Backed up file "Fit"
2023-01-19 19:27:09.268 -05:00 [INF] [@Format] Backed up file "Fit"
2023-01-19 19:27:09.295 -05:00 [INF] [@Format] Backed up file "Fit"
2023-01-19 19:27:09.309 -05:00 [INF] [@Format] Backed up file "Fit"
2023-01-19 19:27:09.337 -05:00 [INF] [@Format] Backed up file "Fit"
2023-01-19 19:27:09.366 -05:00 [INF] [@Format] Backed up file "Fit"
2023-01-19 19:27:09.405 -05:00 [INF] [@Format] Backed up file "Fit"
2023-01-19 19:27:09.463 -05:00 [INF] Done
2023-01-19 19:27:09.466 -05:00 [INF] Sleeping for 86400 seconds...

**Comment by philosowaffle** [comment]
Can you share your config file? (remember to delete your email and passwords)

**Reply by mkblm** [reply]
I didn't notice that the upload to garmin setting was set to false :) Thanks for the very quick response though!

**Reply by mkblm** [reply]
Sorry- another question. Now I'm getting an unauthorized from the Garmin API:
[19:50:24 INF] GUpload: 2023-01-19 19:50:24,233 [CRITICAL] Login Failure: Authentification failed.

I've tripled checked that my email/password are correct in the config file (can successfully login to https://connect.garmin.com/. Any thoughts why I would be getting unauthorized? Is there some setting on my Garmin account re: API access? I tried UploadStrategy 1 as well, same issue.

**Reply by philosowaffle** [reply]
Did a quick sanity check on my end and not able to reproduce with the 3.3.1 windows exe.

If you still have your previous version of P2G in a different folder, try running it and see if you get the same auth error.

If you do get the same auth error even on your previous version, Garmin may have "flagged" P2G and is temporarily blocking it.  The temporary workaround would be to set [SaveLocalCopy](https://github.com/philosowaffle/peloton-to-garmin/blob/master/configuration.example.json#L14) to `true` and manually upload your FIT files to Garmin.

If you do not get the same auth error on the previous version of P2G then can you try switching logging level to [Debug](https://github.com/philosowaffle/peloton-to-garmin/blob/master/configuration.example.json#L51) which may yield a more helpful error message than "Login Failure".

**Comment by mkblm** [comment, marked as answer]
Ugh nevermind, I just noticed that the upload to garmin setting was false by default :/

**Reply by ClusterPuck2** [reply]
I have the same problem.  So is the solution, just to change the line to true?

**Reply by ClusterPuck2** [reply]
That was it, it worked.
